libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -396,20 +397,25 @@ void CodeTransform::operator()(Identifier const& _identifier)
 void CodeTransform::operator()(Literal const& _literal)
 {
 	m_assembly.setSourceLocation(_literal.location);
-	if (_literal.kind == LiteralKind::Number)
-		m_assembly.appendConstant(u256(_literal.value.str()));
-	else if (_literal.kind == LiteralKind::Boolean)
-	{
-		if (_literal.value == "true"_yulstring)
-			m_assembly.appendConstant(u256(1));
-		else
-			m_assembly.appendConstant(u256(0));
-	}
-	else
+
+	u256 constant;
+	switch(_literal.kind)
```

Comment:
This is exactly ``valueOfLiteral`` now, isn't it?

Reply:
The asserts types differ, but yes

Reply:
I don't think the assert types matter too much here.

Reply:
I would prefer having it all in one function, too

Reply:
I think we should use the code, even if it might throw the wrong assertion exception types.